### PR TITLE
Catch step exceptions

### PR DESCRIPTION
**Pull request by changlong-liu**
fix issue #328: clear internal resources even when test failed

### CODE REVIEW

File: src/test/scenarios/attestation/output/src/attestation/azext_attestation/tests/__init__.py [hunk 1]
```diff
@@ -34,16 +37,34 @@ def get_func_to_call():
...
         return wrapper
+
+
+def raise_if():
+    if exceptions:
+        if len(exceptions) <= 1:
+            raise exceptions[0][1][1]
+        message = "{}\nFollowed with exceptions in other steps:\n".format(str(exceptions[0][1][1]))
+        message += "\n".join(["{}: {}".format(h[0], h[1][1]) for h in exceptions[1:]])
+        raise exceptions[0][1][0](message).with_traceback(exceptions[0][1][2])
```
Comment (qiaozha) [inline src/test/scenarios/attestation/output/src/attestation/azext_attestation/tests/__init__.py:70]: Is exceptions a three demensions array? or one demension but with dictionary type element ?

Comment (changlong-liu) [reply src/test/scenarios/attestation/output/src/attestation/azext_attestation/tests/__init__.py:70]: refer to Line 66:
~~~
output.push('            exceptions.append((func.__name__, sys.exc_info()))');
~~~

The exceptions is an array of (step_name, exception_info), where step_name is a string,  and exception_info is a 3 elements tuple: 
exception_info -> (exception_type, exception_instance, stack_info)

Comment (qiaozha) [reply src/test/scenarios/attestation/output/src/attestation/azext_attestation/tests/__init__.py:70]: okay. Is exception_instance a string type? 
`message = "{}\nFollowed with exceptions in other steps:\n".format(str(exceptions[0][1][1]))`

Comment (changlong-liu) [reply src/test/scenarios/attestation/output/src/attestation/azext_attestation/tests/__init__.py:70]: No, it should be some subclass of BaseException

Comment (qiaozha) [reply src/test/scenarios/attestation/output/src/attestation/azext_attestation/tests/__init__.py:70]: Then are you sure this line work 
`message = "{}\nFollowed with exceptions in other steps:\n".format(str(exceptions[0][1][1]))`

**Pull request merged**